### ISSUE TITLE
Current implementation of output custom modes breaks the protocol

### ISSUE THREAD

**Issue by dezeroku** [swaywm/sway#7868]
- **Sway Version:**
  - `sway version 1.8.1`
  - looking at the source code also exists on `master`

- **Debug Log:**
  - [whole log (github gist)](https://gist.github.com/dezeroku/89503a1b9ed0c19f84c87cf716f90344)
  - the important part:
```00:00:18.614 [DEBUG] [sway/config/output.c:223] Adding non-wildcard output config
00:00:18.614 [DEBUG] [sway/config/output.c:226] Merging on top of output * config
00:00:18.614 [DEBUG] [sway/config/output.c:240] Config stored for output eDP-1 (enabled: 1) (1280x720@60.000000Hz position 0,0 scale 1.000000 subpixel unknown transform 0) (bg /usr/share/backgrounds/sway/Sway_Wallpaper_Blue_1920x1080.png fill) (power -1) (max render time: -1)
00:00:18.614 [DEBUG] [sway/config/output.c:395] Turning on output eDP-1
00:00:18.614 [DEBUG] [sway/config/output.c:403] Set eDP-1 mode to 1280x720 (60.000000 Hz)
00:00:18.614 [ERROR] [sway/config/output.c:278] Configured mode for eDP-1 not available
00:00:18.614 [INFO] [sway/config/output.c:279] Picking preferred mode instead
```

- **Configuration File:**
  - Reproduced with Arch Linux's default config

- **Stack Trace:**
  - N/A

- **Description:**

I've been trying to add support for custom output modes to [kanshi](https://todo.sr.ht/~emersion/kanshi/80) project and ended up with the same implementation as in [this MR](https://github.com/emersion/kanshi/pull/78/files).

The important part here is the call to `set_custom_mode` defined in [wlr-output-management-unstable-v1.xml](https://gitlab.freedesktop.org/wlroots/wlr-protocols/-/blob/master/unstable/wlr-output-management-unstable-v1.xml?ref_type=heads#L556) protocol.

The call goes through, but fails in sway with the `Configured mode for eDP-1 not available` message and falls back to the preferred mode.

How I reproduced it locally:
- start up sway
- start up kanshi with the modified config using `--custom`

This can be definitely be minimized to a short C file calling the `set_custom_mode`, but my C and Wayland knowledge is not good enough to write it. I'll try to do it anyway and get back here

**Debugging**

The `set_custom_mode` request is received by sway, but not properly handled. It should enter [the if clause in sway/config/output.c]( https://github.com/swaywm/sway/blob/master/sway/config/output.c#L257), but for this to happen the `custom` boolean must be truthy. The `custom` boolean is set in [`merge_output_config` function](https://github.com/swaywm/sway/blob/master/sway/config/output.c#L110) in the same file:
```
	if (src->custom_mode != -1) {
		dst->custom_mode = src->custom_mode;
	}
```
The only place in the project where `custom_mode` is set to truthy is in the [output command's source code](https://github.com/swaywm/sway/blob/master/sway/commands/output/mode.c#L18) and I don't think we can enter it over a protocol.

To confirm this I've modified the sway config to include the following line:
```
output * mode --custom 1920x1080
```
This flips the boolean in the config and then `set_custom_mode` call over protocol works, but doing it this way feels very clunky.

To summarize, I understand that this check was added to prevent users from using custom mode unknowingly (without the `--custom` argument) with the `swaymsg "output ..."` command, but it also seems to block setting custom modes over the protocol itself.
